File: app/src/main/java/com/arbaelbarca/listmovieskatalog/database/room/FavoriteDao.java
package com.arbaelbarca.listmovieskatalog.database.room;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.arbaelbarca.listmovieskatalog.model.movies.ResultsItem;

import java.util.List;

@Dao
public interface FavoriteDao {
    @Query("SELECT * FROM favoritemoviesnew")
    Cursor selectAll();

    @Query("SELECT * FROM favoritemoviesnew where id = :id")
    Cursor selectById(long id);

    @Insert
    public void addData(ResultsItem favoriteList);

    @Query("select * from favoritemoviesnew")
    public List<ResultsItem> getFavoriteData();

    @Query("SELECT EXISTS (SELECT 1 FROM favoritemoviesnew WHERE id=:id)")
    public int isFavorite(int id);

    @Delete
    public void delete(ResultsItem favoriteList);

}

File: app/src/main/java/com/robytakwa/listpicturekatalog/database/room/FavoriteDao.kt
package com.robytakwa.listpicturekatalog.database.room

import android.database.Cursor

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

import com.robytakwa.listpicturekatalog.model.movies.ResultsItem

@Dao
interface FavoriteDao {

    @get:Query("select * from favoritemoviesnew")
    val favoriteData: List<ResultsItem>

    @Query("SELECT * FROM favoritemoviesnew")
    fun selectAll(): Cursor

    @Query("SELECT * FROM favoritemoviesnew where id = :id")
    fun selectById(id: Long): Cursor

    @Insert
    fun addData(favoriteList: ResultsItem)

    @Query("SELECT EXISTS (SELECT 1 FROM favoritemoviesnew WHERE id=:id)")
    fun isFavorite(id: Int): Int

    @Delete
    fun delete(favoriteList: ResultsItem)

}
